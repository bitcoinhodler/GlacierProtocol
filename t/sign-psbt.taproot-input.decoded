{
  "txid": "6128af8903fb97bd718944510eef667c7b1d6d52bc7031166782023ecd2f5234",
  "hash": "cf123f0d289eb9b4e40c50e80cfbf5db67d3e9c1c60ed3fc1029e7941d538053",
  "version": 2,
  "size": 382,
  "vsize": 166,
  "weight": 664,
  "locktime": 0,
  "vin": [
    {
      "txid": "5e1110500f6424350edf718a95650699b71f6330e951de0ea847926b8702c360",
      "vout": 1,
      "scriptSig": {
        "asm": "",
        "hex": ""
      },
      "txinwitness": [
        "",
        "3044022003321f02c6618e6a28e24975c5e4c539c31e105b6966cb1263f047c365dc60c302206f5b485fc87f0e0083b24cced0a965819ef68ce9438332b6e781b2bc468b9b6801",
        "304402207d0d82b7d8f87000a47b9014122f60aaf8368e5ec5da61efa171437b8d80fdce022077e157f7debaa5b9e6a24721634d8b4aa104d3c57aefb772cd44c35049219c9101",
        "522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.00009834,
      "n": 0,
      "scriptPubKey": {
        "asm": "1 901069f8e009bb67a27b47054150c60287b0af5511bb214facfebe0945114df4",
        "desc": "rawtr(901069f8e009bb67a27b47054150c60287b0af5511bb214facfebe0945114df4)#4chsgtwg",
        "hex": "5120901069f8e009bb67a27b47054150c60287b0af5511bb214facfebe0945114df4",
        "address": "bcrt1pjqgxn78qpxak0gnmguz5z5xxq2rmpt64zxajznavl6lqj3g3fh6qlvaknf",
        "type": "witness_v1_taproot"
      }
    }
  ]
}
