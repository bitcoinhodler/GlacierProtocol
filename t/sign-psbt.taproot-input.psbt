cHNidP8BAF4CAAAAAWDDAodrkkeoDt5R6TBjH7eZBmWVinHfDjUkZA9QEBFeAQAAAAD/////AWomAAAAAAAAIlEgkBBp+OAJu2eie0cFQVDGAoewr1URuyFPrP6+CUURTfQAAAAAAAEBKxAnAAAAAAAAIgAgZchjx1AzHMA++02sohfnklz/h/XpypknVkI0vg1ieOgBBYtSIQPRTdz7aBf1V5aVu7PrPhhYh78pQrAx5vcWNDuP5+no4iECj81G+GFLLL8xgJaWgkKh4ivPttjyttyTnIwnw0eyk3shAxWstVASD0zctGDVxJCAq1CMpMzY3t7Kwi/urIzQF9TBIQIrBj7i8i+eGYLBQP53hnLGgxEaeW3Mzb1Hu2Xj1gipg1SuAAA=
