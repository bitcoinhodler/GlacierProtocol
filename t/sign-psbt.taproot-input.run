#!/bin/bash
set -e

# Test withdrawing from transaction that also has a taproot output.
# We send our withdrawal to a taproot output too.

$GLACIERSCRIPT --regtest=$1 sign-psbt << INPUT 2>&1
y
y
y
y
y
y
../../t/sign-psbt.taproot-input.psbt
y
2
cNYaH3onqrdMffpznhMMmrHn34fuTU59w5j8LM3H42VPcUsLeXy5
cRoydfinDRzzRQJp5niqJWukSYTfPJQM6ytqGN6nzonaz1mafgwD
y
INPUT
