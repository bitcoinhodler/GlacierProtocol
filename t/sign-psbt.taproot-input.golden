Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?Input a filename located in the current directory which contains the PSBT:
0.00010000 BTC in unspent inputs from cold storage address bcrt1qvhyx836sxvwvq0hmfkk2y9l8jfw0lpl4a89fjf6kgg6turtz0r5q2wgvlj
0.00009834 BTC going to destination address bcrt1pjqgxn78qpxak0gnmguz5z5xxq2rmpt64zxajznavl6lqj3g3fh6qlvaknf
Fee amount: 0.00000166
Confirm? (y/n): 
How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: 
Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.00010000 BTC in unspent inputs from cold storage address bcrt1qvhyx836sxvwvq0hmfkk2y9l8jfw0lpl4a89fjf6kgg6turtz0r5q2wgvlj
0.00009834 BTC going to destination address bcrt1pjqgxn78qpxak0gnmguz5z5xxq2rmpt64zxajznavl6lqj3g3fh6qlvaknf
Fee amount: 0.00000166

Signing with private keys: 
cNYaH3onqrdMffpznhMMmrHn34fuTU59w5j8LM3H42VPcUsLeXy5
cRoydfinDRzzRQJp5niqJWukSYTfPJQM6ytqGN6nzonaz1mafgwD


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 1.00 satoshis per vbyte

Raw signed transaction (hex):
0200000000010160c302876b9247a80ede51e930631fb7990665958a71df0e3524640f5010115e0100000000ffffffff016a26000000000000225120901069f8e009bb67a27b47054150c60287b0af5511bb214facfebe0945114df40400473044022003321f02c6618e6a28e24975c5e4c539c31e105b6966cb1263f047c365dc60c302206f5b485fc87f0e0083b24cced0a965819ef68ce9438332b6e781b2bc468b9b680147304402207d0d82b7d8f87000a47b9014122f60aaf8368e5ec5da61efa171437b8d80fdce022077e157f7debaa5b9e6a24721634d8b4aa104d3c57aefb772cd44c35049219c91018b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae00000000

Transaction fingerprint (md5):
c42aad7ec9d4045adaa42793355065ee
QR code for transaction written to transaction.png
